ProcessLib/DensityDrivenFlow/DensityDrivenFlowFEM.h

```diff
@@ -0,0 +1,250 @@
+
+    using LocalAssemblerTraits =
+        ProcessLib::LocalAssemblerTraits<ShapeMatricesType,
+                                         ShapeFunction::NPOINTS, NUM_NODAL_DOF,
+                                         /* number of pcs vars */ GlobalDim>;
+    using NodalMatrixType = typename LocalAssemblerTraits::LocalMatrix;
+    using NodalVectorType = typename LocalAssemblerTraits::LocalVector;
+    using GlobalDimVectorType = typename ShapeMatricesType::GlobalDimVectorType;
+
+public:
+    LocalAssemblerData(MeshLib::Element const& element,
+                       std::size_t const local_matrix_size,
+                       bool is_axially_symmetric,
+                       unsigned const integration_order,
+                       DensityDrivenFlowProcessData const& process_data)
+        : _element(element),
+          _process_data(process_data),
+          _integration_method(integration_order),
+          _shape_matrices(initShapeMatrices<ShapeFunction, ShapeMatricesType,
+                                            IntegrationMethod, GlobalDim>(
+              element, is_axially_symmetric, _integration_method)),
+          _darcy_velocities(
+              GlobalDim,
+              std::vector<double>(_integration_method.getNumberOfPoints()))
+    {
+        // This assertion is valid only if all nodal d.o.f. use the same shape
+        // matrices.
+        assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
+        (void)local_matrix_size;
+    }
+
+    void assemble(double const t, std::vector<double> const& local_x,
+                  std::vector<double>& local_M_data,
+                  std::vector<double>& local_K_data,
+                  std::vector<double>& local_b_data) override
+    {
+        auto const local_matrix_size = local_x.size();
+        // This assertion is valid only if all nodal d.o.f. use the same shape
+        // matrices.
+        assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
+
+        auto local_M = MathLib::createZeroedMatrix<NodalMatrixType>(
+            local_M_data, local_matrix_size, local_matrix_size);
+        auto local_K = MathLib::createZeroedMatrix<NodalMatrixType>(
+            local_K_data, local_matrix_size, local_matrix_size);
+        auto local_b = MathLib::createZeroedVector<NodalVectorType>(
+            local_b_data, local_matrix_size);
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        SpatialPosition pos;
+        pos.setElementID(_element.getID());
+
+        for (std::size_t ip(0); ip < n_integration_points; ip++)
+        {
+            pos.setIntegrationPoint(ip);
+            auto const temperature0 =
+                _process_data.reference_temperature(t, pos)[0];
+            auto const beta =
+                _process_data.thermal_expansion_coefficient(t, pos)[0];
+
+            auto const body_force = _process_data.specific_body_force(t, pos);
```

Comment:
I mean a constant vector. I think it is the gravity vector, since the density is not included in the input. If it is called   body_force, one may make a mistake by inputting (0, 0, rho*g) for it. 

Since rho can vary, that type of input for LiuidFlowProcess just means 'gravity vector component  in i direction'.